### ISSUE TITLE
mssql database name with dash fails

### ISSUE THREAD

**Issue by amitport** [typeorm/typeorm#1486]
when trying to setup a connection with a database with a dash (`-`, e.g., `DEV-MY-DB`) the connection fails with:

QueryFailedError: Error: Incorrect syntax near '-'.
    at new QueryFailedError (<my-project>\src\error\QueryFailedError.ts:7:9)
    at <my-project>\src\driver\sqlserver\SqlServerQueryRunner.ts:320:37
    at _query (<my-project>\node_modules\mssql\lib\base.js:1253:25)
    at Request.tds.Request.err [as userCallback] (<my-project>\node_modules\mssql\lib\tedious.js:620:15)
    at Request._this.callback <my-project>\node_modules\tedious\lib\request.js:60:27)
    at Connection.endOfMessageMarkerReceived (<my-project>\node_modules\tedious\lib\connection.js:1760:27)
    at Connection.dispatchEvent (<my-project>\node_modules\tedious\lib\connection.js:857:45)
    at Parser.<anonymous> <my-project>\node_modules\tedious\lib\connection.js:683:18)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)

**Comment by pleerock** [comment]
can you please show the failed query itself? (set logging: true in connection options)

**Comment by amitport** [comment]
```
executing query:  SELECT * FROM MY-DB.INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME IN ('XN_user') AND TABLE_SCHEMA IN (SCHEMA_NAME())
query failed:  SELECT * FROM MY-DB.INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME IN ('XN_user') AND TABLE_SCHEMA IN (SCHEMA_NAME())
error:  { RequestError: Incorrect syntax near '-'.
    at handleError (<my-project>\node_modules\mssql\lib\tedious.js:519:15)
    at emitOne (events.js:116:13)
    at Connection.emit (events.js:211:7)
    at Parser.<anonymous> (<my-project>\node_modules\tedious\lib\connection.js:489:16)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)
    at Parser.<anonymous> (<my-project>\node_modules\tedious\lib\token\token-stream-parser.js:54:15)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)
    at addChunk (<my-project>\node_modules\readable-stream\lib\_stream_readable.js:284:12)
  code: 'EREQUEST',
  number: 102,
  lineNumber: 1,
  state: 1,
  class: 15,
  serverName: '<my-mssql-server>',
  procName: '',
  originalError: 
   { Error: Incorrect syntax near '-'.
    at handleError (<my-project>\node_modules\mssql\lib\tedious.js:517:19)
    at emitOne (events.js:116:13)
    at Connection.emit (events.js:211:7)
    at Parser.<anonymous> (<my-project>\node_modules\tedious\lib\connection.js:489:16)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)
    at Parser.<anonymous> (<my-project>\node_modules\tedious\lib\token\token-stream-parser.js:54:15)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)
    at addChunk (<my-project>\node_modules\readable-stream\lib\_stream_readable.js:284:12)
     info: 
      { number: 102,
        state: 1,
        class: 15,
        message: 'Incorrect syntax near \'-\'.',
        serverName: '<my-mssql-server>',
        procName: '',
        lineNumber: 1,
        name: 'ERROR',
        event: 'errorMessage' } },
  name: 'RequestError',
  precedingErrors: [] }
Debugger attached.
[Nest] 10320   - 2018-1-23 09:05:34   [ExceptionHandler] Error: Incorrect syntax near '-'.
QueryFailedError: Error: Incorrect syntax near '-'.
    at new QueryFailedError (<my-project>\src\error\QueryFailedError.ts:7:9)
    at <my-project>\src\driver\sqlserver\SqlServerQueryRunner.ts:320:37
    at _query (<my-project>\node_modules\mssql\lib\base.js:1253:25)
    at Request.tds.Request.err [as userCallback] (<my-project>\node_modules\mssql\lib\tedious.js:620:15)
    at Request._this.callback (<my-project>\node_modules\tedious\lib\request.js:60:27)
    at Connection.endOfMessageMarkerReceived (<my-project>\node_modules\tedious\lib\connection.js:1760:27)
    at Connection.dispatchEvent (<my-project>\node_modules\tedious\lib\connection.js:857:45)
    at Parser.<anonymous> (<my-project>\node_modules\tedious\lib\connection.js:683:18)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)
```

**Comment by pleerock** [comment]
okay, it needs escaping. We already fixed this issue and it will be released in 0.2.0 in a few months. Can you make the changes on your local until that? Or make a PR with fix (easy)?

**Comment by amitport** [comment]
I'm not sure where to make a PR. In any case, this isn't really an urgent issue for me- I just changed the DB name :).

**Comment by pleerock** [comment]
ah okay, that sounds good

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
